### PR TITLE
Feature/frontend/mods/dropdown

### LINKED ISSUE #18: Use version dropdown in "manage mods" view
image

### PR DESCRIPTION
**Pull request by danpolanco**
Double check this for me?

Seems fragile because (a) I'm reusing code via copy and paste and (b) I removed `if (_.includes(cModdedTypes, newValue))`. I believe it works though!

Closes #18.

### CODE REVIEW

File: src/main/resources/static/js/mods.js [hunk 1]
```diff
@@ -27,6 +27,19 @@ angular.module('Mccy.mods',[
             $scope.compatibleServerTypes = $scope.mod.serverTypes.map(function(envType){
                 return $filter('serverTypeLabel')(envType);
             }).join(', ');
+
+            $scope.applicableVersions = Versions.query({type:$scope.mod.serverTypes[0]});
```
Comment (itzg) [inline src/main/resources/static/js/mods.js]: This and the `$scope.update` usage are redundant. I like your idea of lazy loading the versions until they actually click to edit a mod's version. So, just get rid of this pre-load of `$scope.applicableVersions`.

File: src/main/resources/static/js/mods.js [hunk 2]
```diff
@@ -1,17 +1,17 @@
 angular.module('Mccy.mods',[
-    'xeditable'
-])
+        'xeditable'
+    ])
 
     .directive('mccyModPanel', function(){
 
-        function controller($scope, $rootScope, $filter, Alerts, Mods) {
+        function controller($scope, $rootScope, $filter, Alerts, Mods, Versions, cModdedTypes) {
```
Comment (itzg) [inline src/main/resources/static/js/mods.js]: `cModdedTypes` doesn't get used, so just remove it for clarity.

File: src/main/resources/static/js/mods.js [hunk 3]
```diff
@@ -34,7 +47,8 @@ angular.module('Mccy.mods',[
             restrict: 'A',
             scope: {
                 mod: '=mccyModPanel',
-                showDelete: '=mccyModPanelShowDelete'
+                showDelete: '=mccyModPanelShowDelete',
+                versions: '=mccyModPanelVersions'
```
Comment (itzg) [inline src/main/resources/static/js/mods.js]: Since you don't need `$scope.versions` from within your directive, just remove this entry.

**Comment by itzg** [comment]
...as for `if (_.includes(cModdedTypes, newValue))`, that was needed in the "new container" controller since it needed to decide between using the full, vanilla list of versions or asking the backend what versions apply to the currently chosen server type. Really, this logic all needs to be encapsulated behind an Angular service...but I'm not yet in the habit of creating those :).

FYI, #39 gives some background about what versions are supported by what type.

File: src/main/resources/static/js/mods.js (hunk 1 above)
Comment (danpolanco) [inline src/main/resources/static/js/mods.js]: Whoops! Forgot to get rid of that O_O

File: src/main/resources/static/js/mods.js (hunk 2 above)
Comment (danpolanco) [inline src/main/resources/static/js/mods.js]: Also forgot to remove. Jeez me :)

**Pull request merged**